src/source-af-packet.c

```diff
@@ -691,7 +693,21 @@ TmEcode AFPWritePacket(Packet *p)
     if (p->afp_v.peer->flags & AFP_SOCK_PROTECT)
         SCMutexLock(&p->afp_v.peer->sock_protect);
     socket = SC_ATOMIC_GET(p->afp_v.peer->socket);
-    if (sendto(socket, GET_PKT_DATA(p), GET_PKT_LEN(p), 0,
+    /* We have a vlan ID without header so first VLAN have been stripped */
+    if (p->vlan_idx && !p->vlanh[0] && p->afp_v.relptr) {
+        pstart = GET_PKT_DATA(p) - VLAN_HEADER_LEN;
+        plen = GET_PKT_LEN(p) + VLAN_HEADER_LEN;
+        /* move ethernet addresses */
+        memmove(pstart, GET_PKT_DATA(p), 2 * ETH_ALEN);
```

Comment:
Does this code get called when we are not in zero copy mode? In that case the reserve call won't do much for us, will it? I worry that GET_PKT_DATA will then point to a suricata-allocated memory block and memmove will write before it.